src/Repository/SearchInputRepository.php

```diff
@@ -0,0 +1,358 @@
+            );
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Input Concept search query and adds it to existing data
+     *
+     * @param $data
+     * @param Concept[] $concepts
+     *
+     * @return array $data
+     */
+    public function generateInputConceptsQuery($data, $concepts)
+    {
+        foreach ($concepts as $concept) {
+            $data[] = [
+                'input' => [
+                    'data' => [
+                        'concepts' => [
+                            [
+                                'name' => $concept->getName(),
+                                'value' => $concept->getValue(),
+                            ],
+                        ],
+                    ],
+                ],
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Output Concept search query and adds it to existing data
+     *
+     * @param $data
+     * @param Concept[] $concepts
+     *
+     * @return array $data
+     */
+    public function generateOutputConceptsQuery($data, $concepts)
+    {
+        foreach ($concepts as $concept) {
+            $data[] = [
+                'output' => [
+                    'data' => [
+                        'concepts' => [
+                            [
+                                'name' => $concept->getName(),
+                                'value' => $concept->getValue(),
+                            ],
+                        ],
+                    ],
+                ],
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Metadata search query and adds it to existing data
+     *
+     * @param $data
+     * @param array $metadata
+     *
+     * @return array $data
+     */
+    public function generateMetadataQuery($data, $metadata)
+    {
+        foreach ($metadata as $searchMetadata) {
+            $data[] = [
+                'input' => [
+                    'data' => [
+                        'metadata' => $searchMetadata,
+                    ],
+                ],
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Image search query and adds it to existing data
+     *
+     * @param $data
+     * @param Input[] $inputs
+     *
+     * @return array $data
+     */
+    public function generateImagesQuery($data, $inputs)
+    {
+        foreach ($inputs as $input) {
+            $data[] = [
+                'input' => [
+                    'data' => [
+                        'image' => [
+                            "url" => $input->getImage(),
+                        ],
+                    ],
+                ],
+            ];
+        }
+
+        return $data;
+    }
+
+    /**
+     * Generates Reverse Image search query and adds it to existing data
+     *
+     * @param $data
+     * @param Input[] $inputs
+     *
+     * @return array $data
+     */
+    public function generateReverseImagesQuery($data, $inputs)
```

Comment:
I've noticed that they are almost the same, but still couldn't figure out how to refactor them. they are using different types of data and sometimes different types of nesting. So it is or 5 almost same methods or one method with more than 5 if(){}'s